### PR TITLE
[Core] Sparse matrices - fix error in centos and skip tests if numpy is not available

### PR DESCRIPTION
**Pull request by ddiezrod**
**📝 Description**
Two things related to latest changes in sparse matrices.
- We were getting an error in centos mpi build as we are trying to bing an lvalue with a &&. 
- I added a skip in two tests that use numpy if it is not available, this gives us some headached in altair CI.

### CODE REVIEW

**Review by RiccardoRossi — COMMENTED** [review]
ok, even though we are now assuming numpy IS available in principle

**Comment by ddiezrod** [comment]
> ok, even though we are now assuming numpy IS available in principle

I know, but the process of installing numpy is still manual in altair side so this will help us until we have dockers in that side too.

**Comment by ddiezrod** [comment]
can you approve it if you are ok? @RiccardoRossi

File: kratos/mpi/tests/test_distributed_sparse_matrices.py [hunk 1]
```diff
@@ -63,9 +67,6 @@ class TestDistributedSparseMatrices(KratosUnittest.TestCase):
         (17,10):1.0,(17,36):1.0,(17,7):1.0,(10,17):1.0,(10,36):1.0,(10,7):1.0,(36,17):1.0,(36,10):1.0,(7,17):1.0,(7,10):1.0,(25,30):1.0,(25,9):1.0,(14,30):1.0,(30,25):1.0,(30,14):1.0,(30,9):1.0,(9,25):1.0,(9,30):1.0
     }
 
-    bref_all = np.array([1,3,2,3,3,3,5,6,4,4,2,4,3,1,6,3,5,3,6,3,0,1,4,1,4,4,6,1,5,2,2,3,1,4,1,5,2,4,1,3])
```
Comment (RiccardoRossi) [inline kratos/mpi/tests/test_distributed_sparse_matrices.py:66]: why did u take out this one?

Comment (ddiezrod) [reply kratos/mpi/tests/test_distributed_sparse_matrices.py:66]: its not being used anywhere and it gives problems as it is not inside the skipped test and it uses numpy.

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
@KratosMultiphysics/technical-committee

**Comment by ddiezrod** [comment]
ping @KratosMultiphysics/technical-committee

**Review by roigcarlo — APPROVED** [review]

**Comment by roigcarlo** [comment]
Just to comment, in principle from ours side is ok, but considering than now numpy is basically a dependency for Kratos, so this dependency in particular should be solved at CI side

**Review by rubenzorrilla — APPROVED** [review]
Approving. In any case still to be discussed in @KratosMultiphysics/technical-committee to establish a clear position about.

**Comment by ddiezrod** [comment]
> Approving. In any case still to be discussed in @KratosMultiphysics/technical-committee to establish a clear position about.

Merging. Anyway, we will add numpy to centos image. It will take some time but it will avoid these problems in the future.

**Comment by rubenzorrilla** [comment]
We discussed in about this in @KratosMultiphysics/technical-committee as well as in  #2165. The position of @KratosMultiphysics/technical-committee is that `numpy` and `h5py` are assumed to be there. Having said this, we accept this change as a temporary solution that must be reverted after you guys add `numpy` to your centos image.

**Pull request merged**